### PR TITLE
MGDCTRS-1793 : Allowing Pod/Containers resource metrics to display in Dataplane Operator Dashboard

### PR DESCRIPTION
**Pull request by rpancham**
## Description
Adding the following metrics to observatorium 
1.kube_pod_info
2.kube_pod_container_resource_requests
3.container_memory_working_set_bytes
4.kube_pod_container_resource_limits
5.kube_pod_container_resource_requests
6.container_network_receive_bytes_total
7.container_network_receive_packets_dropped_total
8.container_network_receive_packets_total
9.container_network_receive_errors_total

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Verified independently by reviewer
- [ ] Required Standard Operating Procedure (SOP) is added.

## Prometheus

- [ ] Changes were tested against dev environment

## Grafana

- [ ] You have imported the changes into a dev or staging environment

### CODE REVIEW

**Review by lgarciaaco — CHANGES_REQUESTED** [review]
You probably need two, max three regex that wrap all the metrics you need. 

If you are going to comment, then the comment should have some meaning

File: resources/prometheus/remote-write.yaml [hunk 1]
```diff
@@ -33,20 +33,126 @@ writeRelabelConfigs:
       - __name__
     targetLabel: __tmp_keep
 
+  # WHAT: Reports the current amount of samples that have tried to be resent to observatorium
+  - action: replace
+    regex: container_cpu_usage_seconds_total*$
```
Comment (shivamG640) [inline resources/prometheus/remote-write.yaml:38]: Not sure why there are 2 entries for the same metrics?

Comment (rpancham) [reply resources/prometheus/remote-write.yaml:38]: @shivamG640  Will have a look on that

**Comment by rpancham** [comment]
@lgarciaaco have added the regex and modified the comment.

File: resources/prometheus/remote-write.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 writeRelabelConfigs:
   # Drop useless labels
   - action: labeldrop
-    regex: endpoint|instance|job|prometheus|pod
+    regex: endpoint|instance|job|prometheus|pod|kube|container
```
Comment (shivamG640) [inline resources/prometheus/remote-write.yaml:4]: I guess, we do use container labels on our dashboard queries!

**Review by shivamG640 — CHANGES_REQUESTED** [review]

File: resources/prometheus/remote-write.yaml [hunk 3]
```diff
@@ -33,20 +33,86 @@ writeRelabelConfigs:
...
+  - action: replace
+    regex: kube_pod_container_resource_limits$
+    replacement: 'true'
+    sourceLabels:
+      - __name__
+    targetLabel: __tmp_keep
+
+  # WHAT: Reports the current amount of samples that have tried to be resent to observatorium
+  - action: replace
+    regex: kube_pod_container_resource_requests$
```
Comment (lgarciaaco) [inline resources/prometheus/remote-write.yaml:62]: You can group all `container_network_.*` metrics in one expression

**Review by lgarciaaco — CHANGES_REQUESTED** [review]
All these metrics are "system" related metrics. Probably the best way to do it is one single expression, something like: `kube_pod_.*|container_network_.*|container_memory_.*`.

File: resources/prometheus/remote-write.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 writeRelabelConfigs:
   # Drop useless labels
   - action: labeldrop
-    regex: endpoint|instance|job|prometheus|pod
+    regex: endpoint|instance|job|prometheus
```
Comment (lgarciaaco) [inline resources/prometheus/remote-write.yaml:4]: Isn't there a way to allow the `pod` label only for the system metrics?

File: resources/prometheus/remote-write.yaml (hunk 3 above)
Comment (rpancham) [reply resources/prometheus/remote-write.yaml:62]: Okay @lgarciaaco i have done this

File: resources/prometheus/remote-write.yaml (hunk 4 above)
Comment (rpancham) [reply resources/prometheus/remote-write.yaml:4]: I am trying to figure out this how this can be resolved. I guess this can be using the action:keep .

Comment (rpancham) [reply resources/prometheus/remote-write.yaml:4]: I have tried to set a  label "pod" for only mentiond metrics and tried it in the local and it is working fine in my local

**Review by lgarciaaco — APPROVED** [review]

**Pull request merged**